src/reducers/toots.js

```diff
@@ -1,33 +1,24 @@
 import { ADD_TOOTS } from '../constants/toots'
+import { accountUrlToInstanceUrl } from '../utils/parser'
 
 const initialState = {}
 
-
 /**
  * @description Update state with new toots
  * @param {Object} state the current state
- * @param {Object} toots An object with items by timelines like follows:
- * { home : [], local : [], federated : []}
+ * @param {Array} toots An array with the toots
+ * @param {String} instanceUrl	The instance url toots are from (eg: https://mastodon.social)
  */
-function addToots(state, { home = [], local = [], federated = [] }) {
-
-	const toots = [...home, ...local, ...federated]
-	const newState = toots.reduce((collection, toot) => {
-		collection[toot.id] = toot
-
-		return collection
-	}, {})
-
-	return {
-		...state,
-		...newState
-	}
+function addToots(state, toots, instanceUrl) {
+	toots = toots.map(toot => [toot.id, toot])
+	const merged = new Map([...state[instanceUrl] || [], ...toots])
```

Comment:
For this kind of data, map > object (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared )

redux-tools should be able to iterate on Maps, it's more a bug from redux-tools